common/network-shuffle/pom.xml

```diff
@@ -58,6 +58,14 @@
       <artifactId>slf4j-api</artifactId>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>com.github.ben-manes.caffeine</groupId>
+      <artifactId>caffeine</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.github.ben-manes.caffeine</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
```

Comment:
Could we avoid this since it depends on Guava and guava conflicts can be a pain to resolve.

Reply:
This is a good idea, but there are some interface differences between Caffeine and Guava Cache, and spark just uses these interfaces. For example, `SessionCatalog` uses `V get(K key, Callable<? extends V> valueLoader)` method only define in Guava Cache API

Reply:
Could you use Caffeine's `V get(K key, Function<? extends K, ? extends V> mappingFunction)` instead? That should be equivalent but be friendly foor method references.

Reply:
Yes, I'm trying to get rid of the dependence on `com.github.ben-manes.caffeine:guava` in this way

Reply:
@holdenk @ben-manes  All the dependencies related to `com.github.ben-manes.caffeine: guava` have been removed.

If all test passed , I will try to upgrade caffeine from 2.9.0 to 2.9.1 in this pr